daffodil-core/src/main/scala/org/apache/daffodil/dsom/ModelGroup.scala

```diff
@@ -44,7 +44,7 @@ object ModelGroupFactory {
 
   /**
    * Because of the contexts where this is used, we return a list. That lets users
-   * flatmap it to get a collection of model groups. Nil for non-model groups, non-Nil for the model group
+   * flatmap it to get a collection of model groups. It will return non-Nil for the model group
    * object. There should be only one non-Nil.
```

Comment:
Is seems like this isn't the case anymore. It looks like it can never return non-Nil.  I think it previously returned Nil when the child passed in was not a model group (eg.. annotation, comment, textNode). But with this change, it's always either a seq, a choice, or a group. So should this now just return a ModelGroup? And callers can Map insead of flatMap?

Reply:
I made updates to both TermFactory and ModelGroup because they should only return a ModelGroup or Term, rather than a list because the callers handle non Model Group/non Term elements. However depending on how the caller handles aforementioned undesirable elements, a flatMap might still be necessary, as is the case with ModelGroup's caller in ComplexType.scala. Undesirable elements in the case are set to Nil, so they can be removed by the flatMap in the caller. So ModelGroupFactory's return still needs to be wrapped in a list. For TermFactory however, this isn't the case, as we remove undesirable elements, rather than doing any processing on them.

Reply:
Ah, your right, the flatMap is still needed. Only suggestion is the switch to an Option for the flatmap since it only flaMaps a Nil or List of one. So replace Nil with None and List(...) with Some(...) Some(...) insead of List. Functionally it's exactly the same thing, but I think more accurately shows that you either have one thing or no things, and the no things get dropped.

Also, in the future, avoid squashing and force-pushing until all comments are address and you get enough +1's for the PR to be accepted. Github doesn't show differences between squashes at all, so it's difficult to figure out what updates have been made to the PR.

+1 :+1:

Reply:
Makes sense! Updates made.